File: 48. Rotate Image/rotate_image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n = len(matrix)
        for i in range(n):
            for j in range(n-1,-1,-1):
                ins_val = matrix[i].pop()
                matrix[j].insert(0,ins_val)

File: RotateImage.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        
        int r=matrix.size();

        for(int i=0;i<r;i++){
            
            for(int j=i+1;j<r;j++){
                
               swap(matrix[i][j],matrix[j][i]);
                
            }
            
        }
        for(int i=0;i<r;i++){
                reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};